src/skmatter/utils/_sparsekde.py

```diff
@@ -0,0 +1,72 @@
+"""The file holds utility functions and classes for the sparse KDE."""
+
+import numpy as np
+
+
+def effdim(cov):
+    """
+    Calculate the effective dimension of a covariance matrix based on Shannon entropy.
+
+    Parameters
+    ----------
+    cov : numpy.ndarray
+        The covariance matrix.
+
+    Returns
+    -------
+    float
+        The effective dimension of the covariance matrix.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skmatter.utils import effdim
+    >>> cov = np.array([[1, 1, 0], [1, 1, 0], [0, 0, 1]])
+    >>> print(round(effdim(cov), 3))
+    1.89
+
+    References
+    ----------
+    https://ieeexplore.ieee.org/document/7098875
+    """
+    eigval = np.linalg.eigvals(cov)
+    eigval /= sum(eigval)
+    eigval *= np.log(eigval)
+    eigval[np.isnan(eigval)] = 0.0
```

Comment:
I feel like this can blow up calculations since it forgives every kind of numerical error. Would like to have some check that the matrix is positive definite
```suggestion
    if (lowest_eigval := np.min(eigval)) <= -np.max(X.shape)*np.finfo(X.dtype).eps:
        raise ValueError(f"Matrix is not positive definite. Lowest eigenvalue {lowest_eigval} is above numerical threshold.")
    eigval[eigval <= 0.] = 0.0
```

Reply:
There is also the cholesky decomposition that performs a positive definite tests, but the logic is buried in some low level language part that I did not read.

Reply:
Very strange, none of the current tests (or examples) passes this check.

Reply:
Maybe it is just a sign flip when you change the global scale in `_covariance` function? Would debug step by step through that function.